Doc/using/windows.rst

```diff
@@ -602,6 +602,49 @@ existed)::
 
     C:\WINDOWS\system32;C:\WINDOWS;C:\Program Files\Python 3.9
 
+.. _win-utf8-mode:
+
+Using UTF-8 mode
+================
+
+.. versionadded:: 3.7
+
+Windows doesn't use UTF-8 for the system encoding (the ANSI Code Page).
+And Python uses the system encoding for the default encoding of text files.
+
+It may cause trouble because UTF-8 is very widely used on the internet
+and most Unix systems, including WSL (Windows Subsystem for Linux).
+
+You can use the UTF-8 mode to change the default text encoding to UTF-8.
+You can enable the UTF-8 mode via ``-X utf8`` command option, or
+``PYTHONUTF8=1`` environement variable.  See :envvar:`PYTHONUTF8` for
+the UTF-8 mode, and :ref:`setting-envvars` for how to modify
+environment variables.
+
+When UTF-8 mode is enabled:
+
+* :func:`locale.getpreferredencoding` returns ``'UTF-8'`` instead of
+  the legacy system encoding.  This function is used for the default
+  text encoding in many places, including :func:`open`, :class:`Popen`,
+  :meth:`Path.read_text`, etc.
+* :data:`sys.stdin`, :data:`sys.stdout`, and :data:`sys.stderr`
+  all use UTF-8 as their text encoding.
+
+.. note::
+   Even when the UTF-8 mode is disabled, Python uses UTF-8 on Windows for:
+
+   * :data:`sys.stdin`, :data:`sys.stdout`, and :data:`sys.stderr`
```

Comment:
In general, it's any console file -- whether opening by fd or name. Even if you open in "C:\\Temp\\con.txt", we resolve this to the actual "\\\\.\\con" path and use `_WindowsConsoleIO` instead of `FileIO`.

Reply:
fixed.